components/ProgressView/src/MDCProgressView.m

```diff
@@ -109,12 +127,105 @@ - (void)setProgressTintColor:(UIColor *)progressTintColor {
     progressTintColor = MDCProgressViewDefaultTintColor();
   }
   self.progressView.backgroundColor = progressTintColor;
+  self.transitionView.backgroundColor = progressTintColor;
 }
 
 - (UIColor *)trackTintColor {
   return self.trackView.backgroundColor;
 }
 
+- (void)setMode:(MDCProgressViewMode)mode {
+  if (_mode == mode) {
+    return;
+  }
+  _mode = mode;
+
+  // If the progress bar was animating in indeterminate mode before, restart the animation.
+  if (_isAnimating && _mode == MDCProgressViewModeIndeterminate) {
+    [self.progressView.layer removeAllAnimations];
+    _isAnimating = false;
+    [self startAnimating];
+  }
+}
+
+- (void)setMode:(MDCProgressViewMode)mode
+       animated:(BOOL)animated
+     completion:(void (^__nullable)(BOOL finished))completion {
```

Comment:
Please check that all early-exit paths call `completion`. I see a few that are missing a call to it.